xray/test/test_backends.py

```diff
@@ -584,6 +584,20 @@ def test_variable_order(self):
         with self.roundtrip(ds) as actual:
             self.assertEqual(list(ds), list(actual))
 
+    def test_unsorted_index_raises(self):
+        random_data = np.random.random(size=(4, 6))
+        dim0 = [0, 1, 2, 3]
+        dim1 = [0, 2, 1, 3, 5, 4]  # We will sort this in a later step
+        da = xray.DataArray(data=random_data, dims=('dim0', 'dim1'),
+                            coords={'dim0': dim0, 'dim1': dim1}, name='randovar')
+        ds = da.to_dataset()
+
+        with self.roundtrip(ds) as ondisk:
+            inds = np.argsort(dim1)
+            ds2 = ondisk.isel(dim1=inds)
+            with self.assertRaisesRegexp(IndexError, 'first by calling .load'),\
+                    self.assertRaisesRegexp(IndexError, 'slices must be sorted'):
+                print(ds2.randovar)  # should raise IndexError in netCDF4
```

Comment:
You want to look at `ds2.randovar.values` to definitely trigger a load from disk. Merely printing it is not enough if the data is in the form of a dask array.